orchestra/actions/install.py

```diff
@@ -93,29 +93,39 @@ def _run(self, explicitly_requested=False):
         self.update_binary_archive_symlink()
 
         post_file_list = self._index_directory(tmp_root + orchestra_root, relative_to=tmp_root + orchestra_root)
-        post_file_list.append(
-            os.path.relpath(installed_component_file_list_path(self.component.name, self.config), orchestra_root)
+        file_list_path = os.path.relpath(
+            installed_component_file_list_path(self.component.name, self.config), orchestra_root
         )
-        post_file_list.append(
-            os.path.relpath(installed_component_metadata_path(self.component.name, self.config), orchestra_root)
+        metadata_path = os.path.relpath(
+            installed_component_metadata_path(self.component.name, self.config), orchestra_root
         )
+        post_file_list.append(file_list_path)
+        post_file_list.append(metadata_path)
         new_files = [f for f in post_file_list if f not in pre_file_list]
 
         if not self.no_merge:
             if is_installed(self.config, self.build.component.name):
                 logger.debug("Uninstalling previously installed build")
                 uninstall(self.build.component.name, self.config)
 
-            logger.debug("Merging installed files into orchestra root directory")
-            self._merge()
-
             self._update_metadata(
                 new_files,
                 install_end_time - install_start_time,
                 source,
                 explicitly_requested,
             )
 
+            logger.debug("Checking for file conflicts")
+            no_conflicts = self._check_conflicts(file_list_path, metadata_path)
+            if not no_conflicts:
+                logger.debug("Conflicting files found, removing metadata")
+                os.remove(os.path.join(orchestra_root, file_list_path))
+                os.remove(os.path.join(orchestra_root, metadata_path))
+                raise UserException("File conflicts detected. Aborting merge")
```

Comment:
1. Can we *not* copy the files instead of removing them after the fact?
2. In case of error we'll get a) `logger.error` from `_check_conflicts`, 2) `logger.debug` from here above, 3) an exception.
  Let's transform `_check_conflicts` into `get_conflicts` and let's emit a single error message.